server/cast/oid.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cast
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/shopspring/decimal"
+
+	"github.com/dolthub/doltgresql/server/functions/framework"
+	pgtypes "github.com/dolthub/doltgresql/server/types"
+)
+
+// initOid handles all explicit and implicit casts that are built-in. This comprises only the "From" types.
+func initOid() {
+	oidExplicit()
+	oidImplicit()
+}
+
+// oidExplicit registers all explicit casts. This comprises only the "From" types.
+func oidExplicit() {
+	framework.MustAddExplicitTypeCast(framework.TypeCast{
+		FromType: pgtypes.Oid,
+		ToType:   pgtypes.BpChar,
+		Function: func(ctx framework.Context, val any, targetType pgtypes.DoltgresType) (any, error) {
+			str := strconv.FormatInt(int64(val.(uint32)), 10)
+			return handleCharExplicitCast(str, targetType)
+		},
+	})
+	framework.MustAddExplicitTypeCast(framework.TypeCast{
+		FromType: pgtypes.Oid,
+		ToType:   pgtypes.Float32,
+		Function: func(ctx framework.Context, val any, targetType pgtypes.DoltgresType) (any, error) {
+			return float32(val.(uint32)), nil
+		},
+	})
+	framework.MustAddExplicitTypeCast(framework.TypeCast{
+		FromType: pgtypes.Oid,
+		ToType:   pgtypes.Float64,
+		Function: func(ctx framework.Context, val any, targetType pgtypes.DoltgresType) (any, error) {
+			return float64(val.(uint32)), nil
+		},
+	})
+	framework.MustAddExplicitTypeCast(framework.TypeCast{
+		FromType: pgtypes.Oid,
+		ToType:   pgtypes.Int16,
+		Function: func(ctx framework.Context, val any, targetType pgtypes.DoltgresType) (any, error) {
+			if val.(uint32) > 32767 {
+				return nil, fmt.Errorf("smallint out of range")
+			}
+			return int16(val.(uint32)), nil
+		},
+	})
+	framework.MustAddExplicitTypeCast(framework.TypeCast{
+		FromType: pgtypes.Oid,
+		ToType:   pgtypes.Int32,
+		Function: func(ctx framework.Context, val any, targetType pgtypes.DoltgresType) (any, error) {
+			return int32(val.(uint32)), nil
```

Comment:
`uint32` has a higher max

Reply:
I added a comment [here](https://github.com/dolthub/doltgresql/pull/204/files#diff-abe27bd98e0177f49ba2b4556c3ae55ed290106afe3c7c32dd6c861beade499eR65) but postgres actually doesn't return an error when casting from oid -> integer. It returns -1 instead 
```
taylor=# SELECT v1::integer FROM t_oid WHERE id=2;
 v1
----
 -1
(1 row)
```